Remove redundant MouseAreas from controls that are Buttons

jarolrod opened the pull request:
For controls that are buttons, an additional `MouseArea` layer is unnecessary as they already have built-in event detection for actions such as [pressing](https://doc.qt.io/qt-6/qml-qtquick-controls2-abstractbutton.html#pressed-prop) and [hovering](https://doc.qt.io/qt-6/qml-qtquick-controls2-control.html#hovered-prop). By eliminating the redundant MouseAreas in the `ContinueButton`, `NavButton`, `OutlineButton`, `Setting`, and `TextButton` controls, we make their implementations cleaner. Previously, the `MouseArea` was utilized to set the controls' state, but now this is achieved through `when` statements that are bundled with the states themselves.


[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/271)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/271)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/271)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/271)

jarolrod commented:
Updated from[ 5fa9522 ](https://github.com/bitcoin-core/gui-qml/commit/5fa9522c692a5190ded6736113ef6b6dd12e5622)to[ b30fefd](https://github.com/bitcoin-core/gui-qml/commit/b30fefd696bc4da420429d911203d77c9ee5b791), [compare](https://github.com/bitcoin-core/gui-qml/compare/5fa9522c692a5190ded6736113ef6b6dd12e5622..b30fefd696bc4da420429d911203d77c9ee5b791)

Changes: updated to fix linter

johnny9 reviewed (changes requested):
`hoverEnabled: true` needs to be set on the buttons for hover to work.

https://doc.qt.io/qt-5/qml-qtquick-controls2-control.html#hoverEnabled-prop

johnny9 commented:
States can be simplified a bit to:
```
       states: [                                                                                                                                                                                                     
            State {                                                                                                                                                                                                   
                name: "PRESSED"; when: root.pressed                                                                                                                                                                   
                PropertyChanges { target: bg; color: Theme.color.orangeLight2 }                                                                                                                                       
            },                                                                                                                                                                                                        
            State {                                                                                                                                                                                                   
                name: "HOVER"; when: root.hovered                                                                                                                                                                     
                PropertyChanges { target: bg; color: Theme.color.orangeLight1 }                                                                                                                                       
            },                                                                                                                                                                                                        
            State {                                                                                                                                                                                                   
                name: "DEFAULT";                                                                                                                                                                                      
                PropertyChanges { target: bg; color: Theme.color.orange }                                                                                                                                             
            }                                                                                                                                                                                                         
        ]               
```

`when` can be thought of like a string of "if-else" statements

jarolrod commented:
Updated from [b30fefd ](https://github.com/bitcoin-core/gui-qml/commit/b30fefd696bc4da420429d911203d77c9ee5b791)to [b86f893](https://github.com/bitcoin-core/gui-qml/commit/b86f89323b346569ab3d7c80876f7b63bad9abfd), [compare](https://github.com/bitcoin-core/gui-qml/compare/b30fefd696bc4da420429d911203d77c9ee5b791..b86f89323b346569ab3d7c80876f7b63bad9abfd)

Changes:
- Simplify when states are triggered per review [feedback](https://github.com/bitcoin-core/gui-qml/pull/271#issuecomment-1439095549)
- Removed the `"DEFAULT"` state as this isn't triggered without a specific condition. When components are not on a state they go back to the definitions given when the component was created, so the `"DEFAULT"` case is now just what where the definitions on creation.
- Reverted the changes to NavButton as that control does need the MouseArea to achieve proper animation of the pressed state


Then updated from [b86f893 ](https://github.com/bitcoin-core/gui-qml/commit/b86f89323b346569ab3d7c80876f7b63bad9abfd)to [f68fd69](https://github.com/bitcoin-core/gui-qml/commit/f68fd695cf968dd45bcf7e2a1fb1784e6ac666fc), [compare](https://github.com/bitcoin-core/gui-qml/compare/b86f89323b346569ab3d7c80876f7b63bad9abfd..f68fd695cf968dd45bcf7e2a1fb1784e6ac666fc)

Changes: 
- Specified `hoverEnabled: true` per review [feedback](https://github.com/bitcoin-core/gui-qml/pull/271#pullrequestreview-1308165747)
- Removed an unnecessary extra condition from the `Setting` control `"HOVER"` state

jarolrod commented:
Updated from[ f68fd69 ](https://github.com/bitcoin-core/gui-qml/commit/f68fd695cf968dd45bcf7e2a1fb1784e6ac666fc)to [80392a5](https://github.com/bitcoin-core/gui-qml/commit/80392a5658b4f4ff975910c4fa8ec57d2b55355b), [compare](https://github.com/bitcoin-core/gui-qml/compare/f68fd695cf968dd45bcf7e2a1fb1784e6ac666fc..80392a5658b4f4ff975910c4fa8ec57d2b55355b)

Changes:
- reversed changes to `Setting` control because it needs a custom mouseArea for the same reasons that `NavButton` needs it
- Ensure that `Setting` and `NavButton` are explicit with `hoverEnabled: true`

johnny9 reviewed (approved):
ACK 80392a5658b4f4ff975910c4fa8ec57d2b55355b
